### PR TITLE
network: introduce Blackhole= setting in [NextHop] section

### PR DESCRIPTION
**Pull request by yuwata**
Split out of #18589.

### CODE REVIEW

File: src/network/networkd-nexthop.c [hunk 1]
```diff
@@ -150,44 +173,58 @@ int manager_get_nexthop_by_id(Manager *manager, uint32_t id, NextHop **ret) {
...
         assert(in);
 
-        existing = set_get(link->nexthops, in);
-        if (existing) {
-                if (ret)
-                        *ret = existing;
-                return 1;
-        }
+        if (link) {
+                existing = set_get(link->nexthops, in);
```
Comment (bluca) [inline src/network/networkd-nexthop.c]: A bit of bikeshedding, but maybe ```set_get(link ? link->nexthops : manager->nexthops, in)``` to reduce indentation and duplication?

File: src/network/networkd-nexthop.c [hunk 2]
```diff
@@ -133,6 +146,16 @@ DEFINE_HASH_OPS_WITH_KEY_DESTRUCTOR(
                 nexthop_compare_func,
                 nexthop_free);
 
+static void nexthop_copy(NextHop *dest, const NextHop *src) {
```
Comment (bluca) [inline src/network/networkd-nexthop.c:149]: this helper doesn't copy all the struct members, so maybe name it something like "nexthop_copy_network_data" to be more specific?

File: src/network/networkd-nexthop.c [hunk 3]
```diff
@@ -204,29 +242,47 @@ static int nexthop_add_internal(Link *link, Set **nexthops, const NextHop *in, N
...
+                                         in, &nexthop);
                 if (r < 0)
                         return r;
                 is_new = true;
         } else if (r == 0) {
                 /* Take over a foreign nexthop */
-                r = set_ensure_put(&link->nexthops, &nexthop_hash_ops, nexthop);
-                if (r < 0)
-                        return r;
+                if (in->blackhole) {
```
Comment (bluca) [inline src/network/networkd-nexthop.c]: maybe ```nexthops = in->blackhole ? &link->manager->nexthops : &link->nexthops;``` etc. to reduce indentation and duplication of the if block?

**Review by bluca — APPROVED** [review]
Just a couple of minor nits, otherwise looks good to me.

File: src/network/networkd-nexthop.c (hunk 2 above)
Comment (yuwata) [reply src/network/networkd-nexthop.c:149]: Instead of rename the function, I added a brief comment.

**Comment by yuwata** [comment]
@bluca Thank you for your review! All comments are addressed. Upgrading the green label.

**Pull request merged**